src/Core/Framework/DataAbstractionLayer/EntityExtension.php

```diff
@@ -29,7 +29,16 @@ public function extendProtections(EntityProtectionCollection $protections): void
     }
 
     /**
+     * @deprecated tag:v6.7.0 - Implement `getEntityName` instead or use `BulkEntityExtension`
      * Defines which entity definition should be extended by this class
      */
     abstract public function getDefinitionClass(): string;
+
+    /**
+     * @deprecated tag:v6.7.0 - reason:visibility-change - Becomes abstract
+     */
+    public function getEntityName(): string
```

Comment:
we should use 
```
Feature::triggerOrThrow();
```
here, as when this method is executed it shows that somewhere there is an extension that relies on deprecated behaviour and we wanna be notified with deprecation notices about that

Reply:
> use triggerOrThrow in getEntityName() only in dev mode

How? I would not "try" to inject the current environemnt into all extensions - IMO too risky.

Reply:
what about:
```php
if (EnvironmentHelper::getVariable('APP_ENV') === 'dev') {
   Feature::triggerOrThrow();
}
```

Reply:
FYI: When you create a new project, The DevOps bundle is not included for production:

![image](https://github.com/user-attachments/assets/b75d80df-c10d-4c50-81c1-3607956de718)

I just implemented it like this, also because it seems that you use it somewhere else - some customers complain right now regarding this issue because you use it somewhere in the ThemeCompiler

Reply:
Good catch that was unexpected 🤔 

But it shouldn't break, as the class is autoloaded and available, only the bundle config and services won't be loaded, but in this case we don't need it for accessing the class directly 

There is even a usage of EnvironmentHelper in the admin controller when rendering the admin index template, so if that usage would make issues, the admin shouldn't load at all